src/renderer/components/badge/badge.tsx

```diff
@@ -16,36 +16,67 @@ export interface BadgeProps extends React.HTMLAttributes<any>, TooltipDecoratorP
 @withTooltip
 @observer
 export class Badge extends React.Component<BadgeProps> {
-  @observable.ref elem: HTMLElement;
-  @observable isExpanded = false;
+  @observable _isExpanded = false;
+  @observable hasHighlightedText = false;
+  @observable.ref elem?: HTMLElement;
+
+  componentWillUnmount() {
+    document.removeEventListener("selectionchange", this.onSelectionChange)
+  }
+
+  @computed get isExpanded() {
+    return this.props.isExpanded ?? this._isExpanded
+  }
 
   @computed get isExpandable() {
-    return this.elem?.clientWidth < this.elem?.scrollWidth;
+    const { scrollWidth, clientWidth, clientHeight, scrollHeight } = this.elem || {}
+    return clientWidth < scrollWidth || clientHeight < scrollHeight
+  }
+
+  @autobind()
+  setRef(elem: HTMLElement) {
+    // This needs to be a seperate function, see: https://github.com/facebook/react/issues/11258
+    this.elem = elem
   }
 
   @autobind()
-  onMouseUp(evt: React.MouseEvent) {
-    const isTextSelected = !!document.getSelection().toString().trim();
-    if (!this.isExpandable || isTextSelected) return; // no action required
-    this.isExpanded = !this.isExpanded;
+  onSelectionChange() {
+    this.hasHighlightedText ||= document.getSelection().toString().length > 0
   }
 
   @autobind()
-  bindRef(elem: HTMLElement) {
-    this.elem = elem;
+  onMouseDown() {
+    this.onSelectionChange() // initial "event" fire on mouse down (for clearing old selections)
+    document.addEventListener("selectionchange", this.onSelectionChange)
+  }
+
+  @autobind()
+  onMouseUp() {
+    document.removeEventListener("selectionchange", this.onSelectionChange)
+
+    if (!this.hasHighlightedText) {
+      this._isExpanded = !this._isExpanded
+    }
+
+    this.hasHighlightedText = false
   }
 
   render() {
-    const { className, label, small, children, isExpanded, ...elemProps } = this.props;
+    const { className, label, small, children, isExpanded: _, ...elemProps } = this.props;
     const classNames = cssNames("Badge", className, {
-      small: small,
-      interactive: this.isExpandable,
-      isExpanded: isExpanded ?? this.isExpanded,
+      small,
+      isExpandable: this.isExpandable,
+    })
+    const labelClass = cssNames("label-content", {
+      isExpanded: this.isExpanded,
     })
+
     return (
-      <div {...elemProps} className={classNames} onMouseUp={this.onMouseUp} ref={this.bindRef}>
-        {label}
-        {children}
+      <div {...elemProps} className={classNames} onMouseUp={this.onMouseUp} onMouseDown={this.onMouseDown} >
+        <div className={labelClass} ref={this.setRef}>
```

Comment:
This wrapper redundant and removed in my PR.

Reply:
Good point, I think I previously added it because of how `position: absolute` works. Should we also remove `position: relative` from the `css`?